store/store_download.go

```diff
@@ -273,6 +380,11 @@ func downloadImpl(ctx context.Context, name, sha3_384, downloadURL string, user
 		return err
 	}
 
+	downloadCtx, cancel := context.WithCancel(ctx)
+	defer cancel()
+
+	tc, downloadCtx := NewWriterWithTimeoutAndContext(ctx, downloadSpeedMeasureWindow, downloadSpeedMin)
```

Comment:
Indeed, I made a mistake when applying earlier suggestions from Michael (this didn't break anything because downloadCtx got re-initialized here). The `downloadCtx, cancel := context.WithCancel(ctx)` is removed now.
NewWriterWithTimeoutAndContext() creates own context WithCancel.